daffodil-runtime1/src/main/scala/org/apache/daffodil/layers/LayerTransformer.scala

```diff
@@ -216,7 +216,7 @@ class JavaIOInputStream(s: InputSourceDataInputStream, finfo: FormatInfo)
     }
   }
 
-  override def available(): Int = 1
+  override def available(): Int = 0
```

Comment:
Does nothing rely on this value? Surprised this didn't cause anything to fail.

Reply:
It works as 1, it works as 0. I changed it to zero because I have no possible rationale why it would always be 1. 
The default behavior for available method, provided by java.io.InputStream is to always return 0.